README.md

```diff
@@ -85,932 +119,667 @@ Unlike many authorization systems where a service controls access to resources i
+The REQUIRED nonce parameter `nonce` MAY be any value. A randomly generated string is RECOMMENDED to provide a unique UCAN, though it MAY also be a monotonically increasing count of the number of links in the hash chain. This field helps prevent replay attacks and ensures a unique CID per delegation. The `iss`, `aud`, and `exp` fields together will often ensure that UCANs are unique, but adding the nonce ensures uniqueness.
 
-This calculation MUST NOT take into account [DID fragment]s. If present, fragments are only intended to clarify which of a DID's keys was used to sign a particular UCAN, not to limit which specific key is delegated between. Use `did:key` if delegation to a specific key is desired.
+The recommended size of the nonce differs by key type. In many cases, a random 12-byte nonce is sufficient. If uncertain, check the nonce in your DID's crypto suite.
 
-``` mermaid
-flowchart RL
-  owner[/Alice\] -. owns .-> resource[(Storage)]
-  executor[/"Compute Service"\] --> del2Aud
-  rootIss --> owner
-
-  executor -. accesses .-> resource
-  rootCap -. references .-> resource
-
-  subgraph root [Root UCAN]
-    rootIss(iss: Alice)
-    rootAud(aud: Bob)
-    rootCap("cap: (Storage, crud/*)")
-  end
+This field SHOULD NOT be used to sign arbitrary data, such as signature challenges. See the [`meta`][Metadata] field for more.
 
-  subgraph del1 [Delegated UCAN]
-    del1Iss(iss: Bob) --> rootAud
-    del1Aud(aud: Carol)
-    del1Cap("cap: (Storage, crud/*)") --> rootCap
-  end
+Here is a simple example.
 
-  subgraph del2 [Final UCAN]
-    del2Iss(iss: Carol) --> del1Aud
-    del2Aud(aud: Compute Service)
-    del2Cap("cap: (Storage, crud/*)") --> del1Cap
-  end
+``` js
+{
+  // ...
+  "nonce": {"/": {"bytes": "bGlnaHQgd29yay4"}}
+}
 ```
 
-In the above diagram, Alice has some storage. This storage may exist in one location with a single source of truth, but to help build intuition this example is location independent: local versions and remote stored copies are eventually consistent, and there is no one "correct" copy. As such, we list the owner (Alice) directly on the resource.
+# Metadata
 
-Alice delegates access to Bob. Bob then redelegates to Carol. Carol invokes the UCAN as part of a REST request to a compute service. To do this, she MUST both provide proof that she has access (the UCAN chain), and MUST delegate access to the discharging compute service. The discharging service MUST check that the root issuer (Alice) is in fact the owner (typically the creator) of the resource. This MAY be listed directly on the resource, as it is here. Once the UCAN chain and root ownership are validated, the storage service performs the write.
+The OPTIONAL `meta` field contains a map of arbitrary metadata, facts, and proofs of knowledge. The enclosed data MUST be self-evident and externally verifiable. It MAY include information such as hash preimages, server challenges, a Merkle proof, dictionary data, etc.
```

Comment:
```suggestion
The OPTIONAL `meta` field contains a map of arbitrary metadata, facts, and proofs of knowledge. The enclosed data SHOULD be self-evident and externally verifiable. It MAY include information such as hash preimages, server challenges, a Merkle proof, dictionary data, etc.
```
It feels like this MUST is untestable, and is more like an instruction to UCAN-generating/encoding implementations than anything that could be included in a conformance test? It might be better phrased as a consuming behavior, i.e., "Validators MUST NOT consider the contents when proofing UCAN invocation".

corner-case: do CIDs in metadata have to be dereferenceable at runtime as well? or does their being unmeaningful trump their being IPLD 😄